lib/puppet/type/zfs_share.rb

```diff
@@ -0,0 +1,234 @@
+Puppet::Type.newtype(:zfs_share) do
```

Comment:
I'm not necessarily opposed to it, though I don't know if it improves upon anything. In that scenario it seems as though there is a greater likelihood that a user would input incorrect parameters, and validation has to move out of the type and into the provider, which does not necessarily seem advantageous. Alternately, validation goes away, which does not help in ensuring that the configuration placed is valid, especially given that Solaris is quite willing to accept a configuration that does not work for some of these parameters.

Given that this type of change would result in another rework of this, and shifting of the validation, I'm inclined to prefer the current module of being verbose with the parameters that can be set.